src/compiler.jl

```diff
@@ -5904,48 +5904,58 @@ base_type(T::UnionAll) = base_type(T.body)
 base_type(T::DataType) = T
 
 # return result and if contains any
-function to_tape_type(Type::LLVM.LLVMType)::Tuple{DataType,Bool}
-    if isa(Type, LLVM.StructType)
+function to_tape_type(Type::LLVM.API.LLVMTypeRef)::Tuple{DataType,Bool}
+    tkind = LLVM.API.LLVMGetTypeKind(Type)
+    if tkind == LLVM.API.LLVMStructTypeKind
         tys = DataType[]
+        nelems = LLVM.API.LLVMCountStructElementTypes(Type)
         containsAny = false
-        for e in LLVM.elements(Type)
+        for i in 1:nelems
+            e = LLVM.API.LLVMStructGetTypeAtIndex(Type, i-1)
             T, sub = to_tape_type(e)
             containsAny |= sub
             push!(tys, T)
         end
         Tup = Tuple{tys...}
         if containsAny
-            return AnonymousStruct(Tup), false
+            return NamedTuple{ntuple(i->Symbol(i), Val(Int(nelems))), Tup}, false
```

Comment:
That shouldn't be faster?

Reply:
now, with the explicit creation of the tuple via an array of symbols, it is many many orders of magnitude faster